packages/overlay/src/Overlay.ts

```diff
@@ -0,0 +1,156 @@
+    OverlayFeatures = OverlayPopover(OverlayFeatures);
+} else {
+    OverlayFeatures = OverlayNoPopover(OverlayFeatures);
+}
+
+type OverlayOptions = {
+    delayed?: boolean;
+    notImmediatelyClosable?: boolean;
+    offset?: number | [number, number]; // supporting multi-axis
+    placement?: Placement;
+    receivesFocus?: 'auto' | 'true' | 'false';
+    trigger?: HTMLElement | VirtualTrigger;
+    type?: 'modal' | 'page' | 'hint' | 'auto' | 'manual';
+};
+
+/**
+ * @element sp-overlay
+ *
+ * @fires sp-opened - announces that an overlay has completed any entry animations
+ * @fires sp-closed - announce that an overlay has compelted any exit animations
+ */
+export class Overlay extends OverlayFeatures implements OverlayBase {
+    public static update(): void {
+        const overlayUpdateEvent = new CustomEvent('sp-update-overlays', {
+            bubbles: true,
+            composed: true,
+            cancelable: true,
+        });
+        document.dispatchEvent(overlayUpdateEvent);
+    }
+
+    public static async open(
+        target: HTMLElement,
+        interaction: TriggerInteractionsV1,
+        content: HTMLElement,
+        options: OverlayOptionsV1
+    ): Promise<() => void>;
+    public static async open(
+        content: HTMLElement,
+        options?: OverlayOptions
+    ): Promise<Overlay>;
+    public static async open(
+        targetOrContent: HTMLElement,
+        interactionOrOptions:
+            | TriggerInteractionsV1
+            | OverlayOptions
+            | undefined,
+        content?: HTMLElement,
+        options?: OverlayOptionsV1
+    ): Promise<Overlay | (() => void)> {
+        const v2 = arguments.length === 2;
+        const overlayContent = content || targetOrContent;
+        const overlay = new Overlay();
+        overlay.dispose = () => {
+            overlay.addEventListener('sp-closed', () => {
+                if (!restored) {
+                    restoreContent();
+                    restored = true;
+                }
+                requestAnimationFrame(() => {
+                    overlay.remove();
+                });
+            });
+            overlay.open = false;
+            overlay.dispose = noop;
+        };
+        let restored = false;
+        const restoreContent = reparentChildren([overlayContent], overlay, {
+            position: 'beforeend',
+            prepareCallback: (el) => {
+                const slot = el.slot;
+                el.removeAttribute('slot');
+                return () => {
+                    el.slot = slot;
+                };
+            },
+        });
+        if (v2) {
+            const options = interactionOrOptions as OverlayOptions;
+            overlay.append(overlayContent);
+            overlay.receivesFocus = options.receivesFocus ?? 'auto';
+            overlay.triggerElement = options.trigger || null;
+            overlay.type = options.type || 'modal';
+            overlay.offset = options.offset || 6;
```

Comment:
Unless we want to disallow `offset === 0`:

```suggestion
            overlay.offset = options.offset ?? 6;
```